### DISCUSSION TITLE
List-KR 필터 클리앙 오작동

### DISCUSSION THREAD

**Discussion by Maccvc** [List-KR/community#35] [General]
현재 맥과 아이폰에서 Adguard 앱을 사용중입니다
Adguard 베이스 필터 + List-KR 를 쓰고 있고
DNS 보호는 Cloudflare dns에 Adguard DNS 필터와 List-KR DNS 필터를 사용중입니다(https://cdn.jsdelivr.net/gh/adguardteam/HostlistsRegistry@main/assets/filter_25.txt)

클리앙이라는 사이트만 들어가면 
광고가 있어야할 자리들에 AD라고 표시된 빈칸이 순간적으로 생겼다가 사라집니다
새로고침하거나 글을 읽고 뒤로가기를 할때마다 깜빡이는 증상이 생기는데

그동안은 사이트의 문제인줄 알았으나 
오늘 우연히 youslist 필터만 적용했을 때는 
이런 증상이 없어진다는 걸 알게되어 이곳에 글 남겨봅니다!

List-KR하나만 쓰는게 가장 빨라서
Youslist를 쓰지 않고 해결하고 싶은데 방법이 있을까요?

**Comment by piquark6046** [comment]
그러한 현상은 오작동이 아니라 Extended CSS 규칙이 빈 공간을 지우는 현상입니다.
Extended CSS 말고 일반 cosmetic 규칙으로 작성하면 이러한 현상은 없어지겠지만, 4개월 전에 발생했던 다른 요소를 숨기는 오작동 때문에 Extended CSS을 유지하고 있습니다.
클리앙의 DOM 구조가 안정되어 YousList의 변경 사항이 없는 것 같아, 일단은 cosmetic 규칙으로 변경해보겠습니다.

**Reply by Maccvc** [reply]
넵 감사합니다 :)
이전 질문글의 댓글들을 보고 
추적 보호 필터를 켜는 김에 
브라우저는 Adguard 베이스 필터, 추적 보호 필터, URL 추적 필터, 소셜 미디어 필터, 방해 요소 필터, List-KR
DNS 보호는 Adguard DNS 필터, List-KR DNS 필터 
이렇게 설정했는데 이외에도 권장하시는 설정법이 있을까요?

**Reply by piquark6046** [reply]
나무위키에 있는 파워링크 광고를 차단하실 생각이 있으시면, [NamuLink 유저스크립트](https://github.com/List-KR/NamuLink)를 설치해야 하므로 [Userscripts](https://apps.apple.com/ko/app/userscripts/id1463298887)를 설치하시고 아래 지침을 따라 해주세요.
1. 설치한 Userscripts를 실행해 별도의 폴더를 지정해 유저스크릡트를 저장할 폴더를 선택해주세요.
2. [이 URL](https://cdn.jsdelivr.net/gh/List-KR/NamuLink@main/NamuLink.user.js)를 길게 누르셔서 다운로드하시고 파일을 그 폴더로 이동하세요.
3. iOS 설정에서 Safari -> 확장 프로그램 -> Userscripts를 클릭하여 활성화 및 모든 웹 사이트에서 허용해 주세요. [^1]
4. 나무위키가 이미 열려있다면, 다시 새로고침해주세요.

애드쉴드 광고를 막으시라면, [microShield 유저스크립트](https://github.com/List-KR/microShield)를 동일한 방법으로 설치하시면 됩니다. 설치 URL은 웹 페이지에 있습니다.
[^1]: AdGuard for iOS Pro를 사용 중이시면 AdGuard도 전부 활성화하셔야 합니다.

**Reply by Maccvc** [reply]
클리앙 관련 필터 업데이트를 기다리겠습니다 !
답변 감사합니다!
항상 감사하게 잘 사용하고 있습니다
좋은 하루 되시길 :)

**Reply by piquark6046** [reply]
해결되었습니다. 앞으료 4시간 이내에 수동으로 업데이트를 다운로드하실 수 있습니다.

**Reply by piquark6046** [reply]
업데이트 이후에 오류가 발생하면, 알려주세요.

**Reply by Maccvc** [reply]
> 업데이트 이후에 오류가 발생하면, 알려주세요.

필터가 업데이트 되었길래 시도해봤는데 아직 증상이 계속되고 있습니다
혹시 몰라 DNS 캐시와 크롬 캐시 모두 제거하였지만 그대로네요

Adguard 프리미엄 사용중이고
macOS 사파리, 크롬 모두 같은 증상입니다
첨부파일 확인 부탁드립니다! 
<img width="851" alt="2" src="https://user-images.githubusercontent.com/127823431/225021181-2f68d616-22fe-4f26-a827-11b36f030a59.png">

https://user-images.githubusercontent.com/127823431/225021562-713f149c-3865-4e6c-aaf6-8c73798df68c.mov

**Reply by piquark6046** [reply]
거기는 제가 클리앙에 로그인 했을 때의 DOM 변화를 보지 못해서 따로 건들지 않았습니다.
만약에 가능하다면 DOM 정보를 전송해주시면 좋을 것 같습니다.
비공개 대화가 필요하시면 `filters@adguard.com`로 전송해주세요.

**Reply by Maccvc** [reply]
> 거기는 제가 클리앙에 로그인 했을 때의 DOM 변화를 보지 못해서 따로 건들지 않았습니다. 만약에 가능하다면 DOM 정보를 전송해주시면 좋을 것 같습니다. 비공개 대화가 필요하시면 `filters@adguard.com`로 전송해주세요.

죄송합니다 ㅜㅜ 이쪽에 대해 전혀 아는 바가 없어서
DOM 변화가 뭘 말씀하시는지 어떤 부분을 전송해드려야하는지 감도 안잡히네요 ㅠㅠ

**Reply by piquark6046** [reply]
아래 Javascript 코드를 브라우저 개발자 도구의 Console에서 실행했을 때 어떻게 나오나요?
```javascript
document.querySelectorAll('.nav_container .aside > *')
```
macOS의 Safari에서 개발자 도구 여는 방법: https://support.apple.com/ko-kr/guide/safari/sfri20948/mac

**Reply by Maccvc** [reply]
> 아래 Javascript 코드를 브라우저 개발자 도구의 Console에서 실행했을 때 어떻게 나오나요?
> 
> ```js
> document.querySelectorAll('.nav_container .aside > *')
> ```
> 
> macOS의 Safari에서 개발자 도구 여는 방법: https://support.apple.com/ko-kr/guide/safari/sfri20948/mac

제대로 한건지 모르겠지만
이렇게 나옵니다!

<img width="549" alt="SCR-20230314-tzxh" src="https://user-images.githubusercontent.com/127823431/225028255-10abe031-3ca8-4938-9064-0f20802d2ffb.png">

**Reply by Maccvc** [reply]
아래는 크롬입니다

<img width="546" alt="크롬" src="https://user-images.githubusercontent.com/127823431/225029085-25410278-1d2a-45c0-b0ed-8d1ef5dbfed0.png">

**Reply by piquark6046** [reply]
패치했습니다.

**Reply by Maccvc** [reply]
고생하셨습니다
추후 또 문의 사항이 생기면 글 남길게요
감사합니다 :)
